src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -659,9 +659,21 @@ private async Task<IEnumerable<AzureSupportCenterInsight>> GetInsightsFromDetect
                 .SelectMany(diagnosticData => ((DetectorCollectionRendering)diagnosticData.RenderingProperties).DetectorIds)
                 .Distinct();
 
+            var allDetectors = await ListDetectorsInternal(context);
+
+            // Check if the detector is of type analysis and if yes, get all the detectors associated with the analysis
+            if (detector.Type == DetectorType.Analysis)
+            {
+                var detectorListAnalysis = allDetectors.Where(detectorResponse => detectorResponse.Metadata.AnalysisTypes != null && detectorResponse.Metadata.AnalysisTypes.Contains(detector.Id)).Select(d => d.Metadata.Id);
```

Comment:
Do you want to ignore casing when checking analysis Types?

Reply:
Valid point